PyAssembler/Assembler.py

```diff
@@ -0,0 +1,326 @@
+        
+        # not clr inc dec jmp jne jz sections
+        if opcode_type == 1:                                
+            self.handle_type_1(translated_words, clean_sec_arg)
+        
+        # mov cmp add sub lea xor or rol
+        elif opcode_type == 2:                              
+           self.handle_type_2(translated_words, words[1].rstrip(','), clean_sec_arg)
+        
+        else:
+            raise TooManyParamsException
+        
+        self.current_address += 2
+
+
+    def translate_line(self, asm_line):
+        """ 
+        Converts an assembly code line to it's binary representation.
+        
+        :param asm_line: A string represents the assembly code line
+        :returns: The binary representation of the assembly line.
+        :rtype: str
+        """
+
+        # Remove spaces from indirect expression, if any.
+        words = asm_line.split('[')
+        if len(words) > 1:
+            asm_line = words[0] + "[" + words[1].replace(" ", "")
+        
+        words = asm_line.split()
+        self.handle_label(words)
+        
+        translated_words = [Assembler.EMPTY_WORD, '']
+        
+        # Section line check. None returns if it is.
+        if self.section_handler(words, translated_words) is None:
+            return translated_words[0]
+        
+        # update opcode bits
+        bin_opcode, opcode_type = Assembler.OPCODES.get(words[0])
+        translated_words[0] = bin_opcode + Assembler.EMPTY_WORD[4:]                   
+        
+        try:
+            self.handle_by_type(words, opcode_type, translated_words)
+        
+        except TooManyParamsException:
+            raise Exception("At line: " + asm_line + " : Invalid assembly code line: Too many paramenters")
+        
+        except NotRegisterArgumentException:
+            raise Exception("At line: " + asm_line + " : First parameter of type-2 line must be a register.")
+        
+        return ''.join(translated_words)
+
+
+    def fix_line(self, damaged_line):
+        
+        # Check if the line is damaged (== not binary)
+        if damaged_line.strip('01') == '':
+            return damaged_line
+
+        else:
+            # Re-write the expression as an arithmetic expression using symbol table
+            for label_tuple in self.sym_tbl:
+                damaged_line = damaged_line.replace(label_tuple, str(self.sym_tbl[label_tuple]))
+            
+            damaged_line = eval(damaged_line)
+            return format(int(damaged_line) & 0xffff, '016b')
+
+
+    def section_handler(self, words, translated_words):
+        
+        if words[0] == Sections.STRING:
+            translated_words[0] = self.convert_str_to_binary(words[1].strip('"'))       # remove " "
+        
+        elif words[0] == Sections.DATA:
+            self.current_address += 2
+            translated_words[0] = format(int(words[1]), '016b')
+        
+        else:
+            return True

```

Comment:
This function's return value is `None` when section is found and `False` if section is not found, which is not optimal